tests/raw_to_standard/icone_test.py

```diff
@@ -0,0 +1,218 @@
+
+
+def test_parse_polyline_null_parameter():
+    test_polyline = None
+    test_coordinates = icone.parse_icone_polyline(
+        test_polyline)
+    expected_coordinates = None
+    assert test_coordinates == expected_coordinates
+
+
+def test_parse_polyline_invalid_data():
+    test_polyline = 'invalid'
+    test_coordinates = icone.parse_icone_polyline(
+        test_polyline)
+    expected_coordinates = []
+    assert test_coordinates == expected_coordinates
+
+
+def test_parse_polyline_invalid_coordinates():
+    test_polyline = 'a,b,c,d'
+    test_coordinates = icone.wzdx_translator.parse_polyline(
+        test_polyline)
+    expected_coordinates = []
+    assert test_coordinates == expected_coordinates
+
+
+# --------------------------------------------------------------------------------Unit test for validate_incident function--------------------------------------------------------------------------------
+def test_validate_incident_valid_data():
+    test_valid_output = {
+        '@id': 'U13631595_202012160845',
+        'updatetime': '2020-12-16T17:18:00Z',
+        'starttime': '2020-12-07T14:18:00Z',
+        'description': 'Road constructions are going on',
+        'creationtime': '2020-12-13T14:18:00Z',
+        'location': {
+            'polyline': '34.8380671,-114.1450650,34.8380671,-114.1450650',
+            'street': 'I-70 N'
+        }
+    }
+    assert icone.validate_incident(test_valid_output) == True
+
+
+def test_validate_incident_missing_required_field_description():
+    test_valid_output = {
+        '@id': 'U13631595_202012160845',
+        'updatetime': '2020-12-16T17:18:00Z',
+        'starttime': '2020-12-07T14:18:00Z',
+        'creationtime': '2020-12-13T14:18:00Z',
+        'location': {
+            'polyline': '34.8380671,-114.1450650,34.8380671,-114.1450650'
+        }
+    }
+    assert icone.validate_incident(test_valid_output) == False
+
+
+def test_validate_incident_invalid_start_time():
+    test_valid_output = {
+        '@id': 'U13631595_202012160845',
+        'updatetime': '2020-12-16T17:18:00Z',
+        'starttime': 'dsafsaf',
+        'description': 'Road constructions are going on',
+        'creationtime': '2020-12-13T14:18:00Z',
+        'location': {
+            'polyline': '34.8380671,-114.1450650,34.8380671,-114.1450650',
+            'street': 'I-70 N'
+        }
+    }
+    assert icone.validate_incident(test_valid_output) == False
+
+
+def test_validate_incident_invalid():
+    test_valid_output = 'invalid output'
+    assert icone.validate_incident(test_valid_output) == False
+
+
+def test_validate_incident_no_data():
+    test_valid_output = None
+    assert icone.validate_incident(test_valid_output) == False
+
+
+@patch('uuid.uuid4')
+def test_generate_standard_messages_from_string(mockuuid):
+    uuid.uuid4 = Mock()
+    uuid.uuid4.side_effect = ['we234de', '23wsg54h']
+    xml_string = """<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
Similar to the expected JSON value, should we abstract this xml to an XML file and load from there to clean this up a bit?